### ISSUE TITLE
non snap firefox version failed test

### ISSUE THREAD

**Issue by hhks43** [david-dick/firefox-marionette#16]
Dear David,
I experienced problems between Firefox::Marionette and snap version of firefox.
I downloaded an older non snap version of firefox (91.9.0), but the module test found some  failed tests:  421-423.
Would you check it.
Best regards Sando
[cpan-ff-marionett-install.txt](https://github.com/david-dick/firefox-marionette/files/8886141/cpan-ff-marionett-install.txt)
r

**Comment by david-dick** [comment]
Hi Sando,

I've added a setup-for-firefox-marionette-build.sh script to have a reliable mechanism for disabling the snap version of firefox in ubuntu, but I'm having some issues replicating your bug with the downloaded 91.9.0 version of firefox.

Here is my version of running the test.

[ubuntu_firefox_91_make_output.txt](https://github.com/david-dick/firefox-marionette/files/8887307/ubuntu_firefox_91_make_output.txt)

The error message you are seeing in the output ("Failed to correctly determine the Firefox process id through the initial connection capabilities") means that when Firefox::Marionette forks and execs the firefox binary, the child pid does not match the pid that firefox self reports via the capabilities call.  I'm not sure what is going on to make that happen.  

Is there anything else you changed in your Ubuntu setup that I need to do to replicate your results?

**Comment by hhks43** [comment]
Dear David,

To avoid snap version of firefox I used :  snap remove firefox .
A firefox binary was installed in the /opt directory under /opt/firefox-ver .
I created a bash script to run the non snap firefox ( /opt/firefox-ver/bin/firefox) . 
This step caused failed tests of Firefox::Marionette.
Instead of script using a symbolic link to run non-snap firefox cured the failed tests. 
It is advisable following the instructions of documentation
Install Firefox from Mozilla builds : https://support.mozilla.org/en-US/kb/install-firefox-linux .
After installing firefox this way I could install the newest Firefox::Marionette module by cpan Firefox::Marionette command.

Thank you David for your examination of log and advice.

Best regards Sandor
